tests/js/common/shell/shell-cluster-collection.js

```diff
@@ -411,11 +412,11 @@ function ClusterCollectionSuite () {
 /// @brief test create
 ////////////////////////////////////////////////////////////////////////////////
 
-    testCreateInvalidShardKeys1 : function () {
```

Comment:
I like the new tests, but why have the ShardKey tests been removed?

Reply:
These tests assumed failures when using something like `_rev`, `_from`, `_to` as shard keys.
But they didn't work. The problem is the tests were written like this:
```
try {
  db._create(...);
} catch (err) {
  assert(...)
}
```
The problem is that the create _worked_ and the catch block was never reached.
So I added a `fail()` after the `db._create(...)` calls to fix this.
And then I found that all the creates worked... So it is possible to use these attributes as shard keys. Probably the tests were remainders from pre-3.0, when something like `_from` and `_to` were so special that they didn't work in many cases.

Reply:
Do we allow to shard by `_rev` really?
This value is changed on update.
We cannot change shard keys
=> We can create a collection that disallows upgrades?

Reply:
Ah i see _rev is silently ignored.
I am good with this PR